homeassistant/components/tradfri.py

```diff
@@ -13,17 +13,18 @@
 
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers import discovery
-from homeassistant.const import CONF_HOST, CONF_API_KEY
+from homeassistant.const import CONF_HOST
 from homeassistant.components.discovery import SERVICE_IKEA_TRADFRI
 
-REQUIREMENTS = ['pytradfri==3.0.2',
+REQUIREMENTS = ['pytradfri==4.0.1',
                 'DTLSSocket==0.1.4',
                 'https://github.com/chrysn/aiocoap/archive/'
                 '3286f48f0b949901c8b5c04c0719dc54ab63d431.zip'
                 '#aiocoap==0.3']
 
 DOMAIN = 'tradfri'
-CONFIG_FILE = 'tradfri.conf'
+GATEWAY_IDENTITY = 'homeassistant'
```

Comment:
Should we be using a randomised string here? According to ggravlingen/pytradfri#90, "[Y]ou can only request a key for the same CLIENT_IDENTITY_PROPOSED (9090) once. If you try the same again it'll respond with '4.00 Bad Request'." This presumable means that the only way to re-add a trådfri gateway to homeassistant if it uses a fixed identity string would be a factory reset, which would lose all paired devices.

Reply:
Agreed, and if the identity string is randomly generated then it should probably also be written into the `tradfri.conf` file so the same identity string can be used if hass is restarted.

Reply:
That file is actually now `.tradfri_psk.conf`, which is perfect since humans don't really care about the identity string.

Still, extra code would be needed to generate, read, write, try and maybe regenerate that identity.  `components/updater.py`  already uses `uuid.uuid4() in a similar way.

Reply:
Yes please on this one, as I accidentally generated an access key yesterday with coap using "homeassistant".. 

....also would make it a lot easier to work with several configs/setups/branches.

Reply:
Feel free to raise a PR to change the behaviour. Or discuss the change on the forums.

Either way, this PR isn't the place to discuss it.